Code/GraphMol/SmilesParse/test.cpp

```diff
@@ -4382,6 +4382,36 @@ void testGithub3967() {
   BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
 }
 
+void testGithub6349() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdInfoLog)
+      << "Testing Github Issue 6349: Different SMARTS input formats lead to different SMILES outputs."
+      << std::endl;
+
+  auto checkSmartsToSmiles = [](const std::string &sma,
+                                const std::string &refSmi) {
+    ROMol *molFromSmarts = SmartsToMol(sma);
```

Comment:
Please use an std::unique_ptr here so that the memory gets automatically released

Reply:
Nice catch! Fixed.